tests/test_signature.py

```diff
@@ -0,0 +1,390 @@
+import pytest
+import os
+import json
+from lib.cuckoo.common.path_utils import path_exists
+from lib.cuckoo.common.abstracts import Signature
+from lib.cuckoo.core.plugins import RunSignatures
+from lib.cuckoo.common.constants import CUCKOO_ROOT
+from lib.cuckoo.core.database import Database
+from lib.cuckoo.core.plugins import register_plugin
+
+class FakeSignatureCallAlways(Signature):
+    name = "FakeSignatureCallAlways"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+        
+class FakeSignatureAPI_Cat(Signature):
+    name = "FakeSignatureAPI_Cat"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_apinames = set(["gethostbyname"])
+    filter_categories = set(["network"]) 
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+        
+class FakeSignatureAPI_Process(Signature):
+    name = "FakeSignatureAPI_Process"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_apinames = set(["gethostbyname"])
+    filter_processnames = set(["powershell.exe"])
+     
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+
+class FakeSignatureCat_Process(Signature):
+    name = "FakeSignatureCat_Process"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_processnames = set(["powershell.exe"])
+    filter_categories = set(["network"])
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+
+class FakeSignatureNonFiltered(Signature):
```

Comment:
This signature is quite filtered, so can we rename it to `FakeSignatureAPI_Cat_Process_With_No_OnCall_Check`?

```suggestion
class FakeSignatureAPI_Cat_Process_With_No_OnCall_Check(Signature):
```